include/libdnf/rpm/reldep.hpp

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_RPM_RELDEP_HPP
+#define LIBDNF_RPM_RELDEP_HPP
+
+#include "sack.hpp"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+
+namespace libdnf::rpm {
+/// @brief Represent a relational dependency from libsolv
+///
+/// @replaces libdnf/dnf-reldep.h:struct:DnfReldep
+/// @replaces libdnf/repo/solvable/Dependency.hpp:struct:Dependency
+/// @replaces hawkey:hawkey/__init__.py:class:Reldep
+class Reldep
+{
+public:
+    enum class CmpType {
+        NONE = 0,
+        GT = (1 << 0),
+        EQ = (1 << 1),
+        GTE = (GT | EQ),
+        LT = (1 << 2),
+        LTE = (LT | EQ)
+    };
+
+    /// @brief Creates a reldep from Char*. If parsing fails it raises std::runtime_error.
+    ///
+    /// @param sack p_sack:...
+    /// @param dependency p_dependency:...
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:Dependency(Sack * sack, const std::string & dependency)
+    Reldep(Sack * sack, const std::string & reldep_string);
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:Dependency(const Dependency & dependency);
+    Reldep(const Reldep & reldep);
+    
+    Reldep(const Reldep && reldep) noexcept;
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:~Dependency();
+    /// @replaces libdnf/dnf-reldep.h:function:dnf_reldep_free(DnfReldep *reldep)
+    ~Reldep();
+
+    bool operator==(const Reldep & other) const noexcept;
+    bool operator!=(const Reldep & other) const noexcept;
+    Reldep & operator=(const Reldep & other) noexcept = default;
+    Reldep & operator=(Reldep && other) noexcept;
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:getName()
+    const char * get_name() const;
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:getRelation()
+    const char * get_relation() const;
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:getVersion()
+    const char * get_version() const;
+
+    /// @replaces libdnf/repo/solvable/Dependency.hpp:method:toString()
+    /// @replaces libdnf/dnf-reldep.h:function:dnf_reldep_to_string(DnfReldep *reldep)
+    const char * to_string() const;
```

Comment:
I looked into libsolv source code. In some cases the return value is allocated in temporary space. And in the libsolv header is: "might alloc tmpspace". Please add this information to the method description.

Reply:
const char * replaced by std::string.